### PR TITLE
SONARJAVA-4756 NumberFormatException in AbstractPrintfChecker.getIndex(String param)

### PR DESCRIPTION
**Pull request by kaufco**

### CODE REVIEW

File: java-checks-test-sources/default/src/main/java/checks/PrintfFailCheck.java [hunk 1]
```diff
@@ -292,5 +292,10 @@ void foo(Calendar c) throws java.io.IOException {
     log4j.error("message: {}" + value, value); // Compliant, reported by S3457
     log4j.error("message:" + value, new Exception()); // Compliant, reported by S3457
     log4j.error("message: {}", value, new Exception()); // Compliant
+
+    String.format("%2147483648$g", 42.0); // Noncompliant {{Not enough arguments to feed formater at index 2147483647: '%2147483648$'.}}
+    String.format("%2147483648g", 42.0); // Compliant
+    String.format("%.2147483648g", 42.0); // Compliant
```
Comment (johann-beleites-sonarsource) [inline java-checks-test-sources/default/src/main/java/checks/PrintfFailCheck.java:298]: Did you mean to omit the `$` between the `8` and the `g` in this line?

File: java-checks/src/main/java/org/sonar/java/checks/AbstractPrintfChecker.java [hunk 2]
```diff
@@ -222,7 +222,11 @@ protected void reportMissingPrevious(MethodInvocationTree mit) {
   }
 
   protected static Integer getIndex(String param) {
-    return Integer.valueOf(param.substring(0, param.indexOf('$')));
+    try {
+      return Integer.valueOf(param.substring(0, param.indexOf('$')));
+    } catch (Exception e) {
```
Comment (johann-beleites-sonarsource) [inline java-checks/src/main/java/org/sonar/java/checks/AbstractPrintfChecker.java:227]: Any reason not to catch a more precise exception, e.g. `NumberFormatException`?

**Review by johann-beleites-sonarsource — COMMENTED** [review]
Looks good, a couple of questions for clarification

File: java-checks/src/main/java/org/sonar/java/checks/AbstractPrintfChecker.java (hunk 2 above)
Comment (kaufco) [reply java-checks/src/main/java/org/sonar/java/checks/AbstractPrintfChecker.java:227]: Yes, but not a good one: I just forgot to change it back after I changed it for debugging purposes :man_facepalming: Fixed.

**Review by johann-beleites-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**